src/App.tsx

```diff
@@ -1,223 +1,180 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import {
+  getTodos,
+  createTodo,
+  deleteTodos,
+} from './api/todos';
+
+import { SortType } from './types/SortType';
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortType, setSortType] = useState<SortType>(SortType.ALL);
+  const [error, setError] = useState('');
+  const [title, setTitle] = useState('');
+  const [selectedId, setSelectedId] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+
+  const filteredTodo = useMemo(() => todos.filter(todo => {
+    switch (sortType) {
+      case SortType.ACTIVE:
+        return !todo.completed;
+      case SortType.COMPLETED:
+        return todo.completed;
+      default:
+        return todo;
+    }
+  }), [todos]);
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (errorMessage) {
+        setError(`${errorMessage}`);
+      }
+    };
+
+    getTodosFromServer(user.id);
+  }, []);
+
+  const getValue = (element: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(element.target.value);
+  };
+
+  const addNewTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim() || !user) {
+      setError(ErrorType.ErrorTitle);
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    try {
+      const newTodo = await createTodo(title, user.id);
+
+      setTodos((prevTodos) => [...prevTodos, newTodo]);
```

Comment:
let's call function here for getting all existing todos from server and update the data in state instead of adding it manually